go/src/koding/kites/kloud/provider/google/stack.go

```diff
@@ -254,6 +257,10 @@ func (s *Stack) ApplyTemplate(c *stack.Credential) (*stack.Template, error) {
 
 	t.Resource["google_compute_instance"] = resource.GCInstance
 
+	t.Variable["koding_stack_id"] = map[string]interface{}{
```

Comment:
How then user will be able to ensure unique naming for other resources than `google_compute_instance`?

Lets say user defines `google_compute_address` resource in order to create EIP for his instances. Each resource in GCP requires unique name. How we handle this without `koding_stack_id`?

Also yes, we should expose some IDs in more generic way as well as we should allow user to provide his own cloud-init. But this is much bigger task and should not change `koding_stack_id` variable in the future.